block/retriever.go

```diff
@@ -108,6 +109,16 @@ func (m *Manager) processNextDABatch(daMetaData *da.DASubmitMetaData) error {
 }
 
 func (m *Manager) fetchBatch(daMetaData *da.DASubmitMetaData) da.ResultRetrieveBatch {
+	// Check DA client
+	if daMetaData.Client != m.dalc.GetClientType() {
```

Comment:
is this the only thing need to be validated on the SL batch? what about namespace?

Reply:
yea good thinking.

Reply:
i added a namespace check for celestia, which i think is the only da using namespace id. 
However, i think namespace id configuration needs to be refactored, since there are multiple params in the config for that, and actually it could be directly derived from chain_id without having to pass it to the full-node or having a config param for that